docs/configuration.md

```diff
@@ -0,0 +1,297 @@
+generate-key --path <SECRET_KEY_PATH>`. Generate as many keys as there
+should be core nodes. Last line looks like this:
+
+> [keygen:INFO] Successfully generated primary key and dumped to secrets/testnet/node4.sk, stakeholder id: 39f2bb9fd75ac348e6c92467e61eb3e1418a394f5a2105f9e014b666, PK (base64): /fbQqCqUPdPImYA2djkGYMpj9HZGkDeKhTc/mLcPG7sdhJSR6Ou0sB3J5OII2VdWTXSHRM1cPdNwZsaNba4rcg==
+
+You should use stakeholder ids as keys in `bootStakeholders`
+map. Values are stakeholders' weights, having all weights equal to 1
+is fine.
+
+The second step is VSS certificates generation. Use `cardano-keygen
+--system-start 0 --configuration-key testnet_public_full generate-vss
+--path <SECRET_KEY_PATH>` to generate VSS certificate. This how it
+looks like:
+
+> JSON: key 75fc2050ea497eb615461d01170679d143fb70c07c0baf3db54c6237, value {"expiryEpoch":5,"signature":"b735325b8f21a033cbe3005c35e4397dd33168c62c05cc8f59e66efbcdeb36b5862d7a324def1ba10e4a79cf0e56c75568c84e6ca28b1a9bb5da65fc6a8cb002","signingKey":"4vHA0HXrmpD6VhhCe44CAJ8IHr9BKpCUtOmGRZu/39vRLJ1z2vJp8h+rdOqb7tJg7Uzf94x0NlM8xHgmhPBecg==","vssKey":"WCECml0Nc9bjjerxHGf2sDfKJIILyDKvjM8zCdfzBmYxcyU="}
+
+**IMPORTANT**: you must pass correct `--configuration-key`, because
+certificate validity depends on `protocolMagic`. So if you are
+generating certificates for staging, use `testnet_staging_full`
+instead.
+
+Put this data into `vssCerts` map. There should be as many VSS
+certificates as there are core nodes (i. e. do it for each secret).
+
+## Our configurations
+
+### `mainnet` configurations
+
+There are several mainnet configurations in `lib/configuration.yaml`
+file for different keys.
+
+* `mainnet_base` configuration serves as a basis for other mainnet
+  configurations and shouldn't be used directly.
+* `mainnet_example_generated` configuration is an example of mainnet
+  configuration where genesis is provded by spec. It shouldn't be used
+  directly, only as an example.
+* `mainnet_full` configuration is what core nodes use in real
+  mainnet. `mainnet_wallet_win64` and `mainnet_wallet_macos64` are
+  almost same, but they have different application name and system tag
+  which matters for update system. They should be used by wallets
+  (i. e. nodes launched with Daedalus).
+* `mainnet_dryrun_full` is like `mainnet_full`, but for
+  staging. `mainnet_dryrun_wallet_win64` and
+  `mainnet_dryrun_wallet_macos64` are for staging wallets.
+
+### `devnet` configuration
+
+There is a configuration called `devnet` which should be used to setup
+clusters for developers or QA to test something. There are some values
+which should be set carefully before using this configuration, they
+depend on particular task:
+
+* `slotDuration`, `k` are parameters of the protocol, set to
+  reasonable defaults, but sometimes we may want to use different
+  values.
+* `networkDiameter`, `mdNoBlocksSlotThreshold`,
+  `recoveryHeadersMessage` depend on `k`, comments are provided in
+  configuration file.
```

Comment:
I agree these probably aren't central, buf if this file _is_ documentation for the configuration, shouldn't they be explained here, primarily?

Reply:
Ok, I described these values (also I realized that `networkDiameter` doesn't depend on `k` and removed it from this line, but described it anyway), but in general the issue is mostly describing genesis configuration, so there will be some TODOs left for new issues (describing genesis configuration is more important now).